pkg/refactor/core/runtime_variables.go

```diff
@@ -0,0 +1,70 @@
+package core
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+// RuntimeVariable are direktiv runtime variables that are hold data, workflows performs getting and setting on these
+// data, RuntimeVariables also preserve state across multiple workflow runs.
+type RuntimeVariable struct {
+	ID uuid.UUID
+
+	NamespaceID uuid.UUID
+	WorkflowID  uuid.UUID
+	InstanceID  uuid.UUID
+
+	Scope string
+
+	Name string
+
+	Size     int
+	Hash     string
+	MimeType string
+	Data     []byte
+
+	CreatedAt time.Time
+	UpdatedAt time.Time
+}
+
+type RuntimeVariablesList []*RuntimeVariable
+
+// RuntimeVariablesStore responsible for fetching and setting direktiv runtime variables from datastore.
+type RuntimeVariablesStore interface {
+	// GetByID gets a single runtime variable from store. if no record found,
+	// it returns core.ErrNotFound error.
+	GetByID(ctx context.Context, id uuid.UUID) (*RuntimeVariable, error)
```

Comment:
I'm not sure this method needs to exist. It's only called internally to your driver. Externally, we always reference via compound key of a name plus a uuid (that can refer to either a namespace, an instance, or a workflow.

Reply:
Discussed in chat.